Assembler.UnitTests/MessageReleaserTests.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Assembler.Base;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+using Moq;
+using NUnit.Framework;
+
+namespace Assembler.UnitTests
+{
+    [TestFixture]
+    public class MessageReleaserTests
+    {
+        private MessageReleaser<BaseMessageInAssembly> _releaser;
+        private Mock<ITimeBasedCache<BaseMessageInAssembly>> _cacheMock;
+
+        private Mock<BaseMessageInAssembly> _messageMock;
```

Comment:
That's an abstract class, it's the same as making a mock implementation

Reply:
I've read about it online, and everyone disagree with you. Mocking have a goal - controling behavior - and you dont use it here. The main problems: 
1. Less readable - it wont be clear why you chose to do that
2. Harder for maintainance - if you will change the class (the tested or the entity) in the future, it might be longer  and harder because of the usage of mocks.

I really think that bad, and a DummyFrame is very neat and indicative

Reply:
I've this mock as a utility now, but I didn't actually implement it, I don't think it's necessary.